WordPress/openverse-frontend
# Fix API endpoints for related media and provider stats

## Fixes



This PR fixes the endpoints to fetch related media, updates to use the new endpoints for provider stats, and removes fallback default provider data that was hardcoded into the repo, which can give false assurances that things are working. 

We'll need to test API changes on the front-end much more closely moving forward. 

Noting that in production, the related API endpoint is still seemingly broken, although my local instance of the API is working. 

## Description



Some additional low-hanging cleanup here:

- Fix audio endpoints from 'audios' to 'audio'
- Remove unused provider collection endpoints
- Add some jsdoc annotations for axios
- Fix error handling of `getProviderStats`, which wouldn't actually catch exceptions
- Start the work to make the Sources page media-type agnostic (it still doesn't actually fetch the audio providers)

## Testing Instructions


- Run locally and observe the provider list on the 'sources' page and in the results filter sidebar is visible.
- Run tests and ensure they work
- Bonus: Run a local instance of the API and change the nuxt.config.js to point to it (don't forget `v1` at the end of the endpoint!). Confirm that locally configured sources are visible.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
